### ISSUE TITLE
Generation does not respect batch size

### ISSUE THREAD

**Issue by jyegerlehner** [ibab/tensorflow-wavenet#153]
WaveNetModel.predict_proba() calls _one_hot(), which is using the batch size, but later it reshapes the output of the net to squash it into n x quantization_channels before the softmax, and then again slices out only a single quantization_channels sized vector of the result tensor. 

This came up because we use the same net to train and generate in test_model.py, and it was cleaner to train on a batch containing one waveform for each "speaker_id", so batch_size=3. But when we get to generation the batch_size is already 3, but generation doesn't respect that setting.

So does anyone know if it's that way for a reason, maybe working around a bug in TF? tf.nn.softmax docs read like you can apply the softmax along any axis of an arbitrary tensor, so it seems like it shouldn't be a barrier to preserving the batch size all the way through. 

I guess I'll either try that, or hack the batch_size on WaveNetModel to set it from 3 to 1 in-between training and generating. But thought if maybe somebody knows why it's that way it could save some trouble.

**Comment by nakosung** [comment]
https://www.tensorflow.org/versions/r0.11/resources/faq.html

According to FAQ, we can have shape of batch_size=None to deal with arbitrary batch size. I think we just need to generate one batch at a time so we can have fixed batch_size = 1 for generator code path. Then batch_size is assigned only by audio_reader.
